lampod/src/jsonrpc/channels.rs

```diff
@@ -8,3 +16,87 @@ pub fn json_list_channels(ctx: &LampoDeamon, request: &json::Value) -> Result<js
     let resp = ctx.channel_manager().list_channel();
     Ok(json::to_value(resp)?)
 }
+
+pub fn json_close_channel(ctx: &LampoDeamon, request: &json::Value) -> Result<json::Value, Error> {
+    log::info!("call for `closechannel` with request {:?}", request);
+    let mut request: request::CloseChannel = json::from_value(request.clone())?;
+    let events = ctx.handler().events();
+    // This gives all the channels with associated peer
+    let channels: response::Channels = ctx
+        .handler()
+        .call(
+            "channels",
+            json::json!({
+            "peer_id": request.node_id,
+            }),
+        )
+        .map_err(|err| {
+            Error::Rpc(RpcError {
+                code: -1,
+                message: format!("{err}"),
+                data: None,
+            })
+        })?;
+    let res = if channels.channels.len() > 1 {
+        // check the channel_id if it is not none, if it is return an error
+        // and if it is not none then we need to have the channel_id that needs to be shut
+        if request.channel_id.is_none() {
+            return Err(Error::Rpc(RpcError {
+                code: -1,
+                message: format!("Channels > 1, provide `channel_id`"),
+                data: None,
+            }));
+        } else {
+            // ctx.channel_manager().close_channel(request.clone());
+            request.clone()
```

Comment:
```suggestion
            request.clone()
```

do we really need to clone this?

Reply:
Oops, my bad